epictrack-web/src/components/insights/Work/Tabs/Staff/Charts/workListing.tsx

```diff
@@ -29,22 +42,60 @@ const WorkList = () => {
   const { roles } = useAppSelector((state) => state.user.userDetail);
   const canEdit = hasPermission({ roles, allowed: [ROLES.EDIT] });
 
-  const { data, error, isLoading } = useGetWorksQuery();
-
-  const works = data || [];
-
   useEffect(() => {
     setPagination((prev) => ({
       ...prev,
-      pageSize: works.length,
+      pageSize: wsData.length,
     }));
-  }, [works]);
+  }, [wsData]);
+
+  const getWorkStaffAllocation = React.useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const workStaffingResult = await workService.getWorkStaffDetails();
+      if (workStaffingResult.status === 200) {
+        setWsData(workStaffingResult.data as WorkStaff[]);
+      }
+    } catch (error) {
+      console.error("Work Staffing List: ", error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    getWorkStaffAllocation();
+  }, []);
+
+  React.useEffect(() => {
+    let columns: Array<MRT_ColumnDef<WorkOrWorkStaff>> = [];
+    if (wsData && wsData.length > 0) {
+      const rolename = "Officer / Analyst";
```

Comment:
I think we have role id coming in the api response. It is always better to go with the ID cz the name might change in future.

Reply:
good suggestion